zebrad/src/components.rs

```diff
@@ -5,6 +5,7 @@
 //! component and dependency injection models are designed to work together, but
 //! don't fit the async context well.
 
+pub mod health;
```

Comment:
```suggestion
#[cfg(feature = "health-endpoint")]
pub mod health;
```

Consider guarding the module declaration in components.rs with `#[cfg(feature = "health-endpoint")]`, and remove the redundant `#[cfg(...)]` guards inside `health.rs` (the file will only be compiled when the feature is enabled).